### PR TITLE
Support spawn-style args

### PR DESCRIPTION
**Pull request by jleeothon**

### CODE REVIEW

**Comment by tacconthommy** [comment]
It might be out of scope for this merge request, but should we add a spec that checks the use of the `Eksekuter` direcly?

File: lib/eksek.rb [hunk 1]
```diff
@@ -8,11 +8,11 @@
 module Kernel
   private
 
-  def eksek cmd, opts = {}, &block
-    Eksekuter.new(cmd, opts).run(&block)
+  def eksek *args, **opts, &block
```
Comment (tacconthommy) [inline lib/eksek.rb]: Would it add any value if the `Eksekuter` would check the sanity of the args?

File: lib/eksekuter.rb [hunk 2]
```diff
@@ -6,9 +6,11 @@
 
 # Class that command execution is delegated to by Eksek.
 class Eksekuter
-  def initialize cmd, opts
-    @cmd = cmd
-    @opts = opts.freeze
+  def initialize *args, **opts
+    @env = args[0].is_a?(Hash) ? args.shift : {}
+    @env = @env.each_with_object({}) { |(k, v), o| o[k.to_s] = v }
```
Comment (tacconthommy) [inline lib/eksekuter.rb]: 💖

Comment (jleeothon) [reply lib/eksekuter.rb]: There's also `@env.transform_keys(&:to_s)` which is equivalent to `@env.transform_keys { |k| k.to_s }`, but it's not available in earlier version of Ruby.

File: lib/eksek.rb (hunk 1 above)
Comment (jleeothon) [reply lib/eksek.rb]: What kind of check do you have in mind? Type checks?

Comment (tacconthommy) [reply lib/eksek.rb]: Yes I had first. However, this might be superfluous since we pass stuff on to Open3 anyway... So feel free to not do it and merge. 🤗

**Pull request merged**